ray_on_golem/server/main.py

```diff
@@ -158,6 +164,24 @@ async def ray_service_ctx(app: web.Application) -> None:
     await ray_service.shutdown()
 
 
+async def reputation_service_ctx(app: web.Application) -> None:
+    reputation_service: ReputationService = app["reputation_service"]
+
+    await reputation_service.start()
+    updaters = [
+        ReputationUpdater(network) for network in ["polygon", "mumbai", "goerli", "holesky"]
+    ]
+    update_tasks = [asyncio.create_task(updater.update()) for updater in updaters]
+
+    yield
+
+    for t in update_tasks:
+        t.cancel()
+
+    await asyncio.gather(*update_tasks)
```

Comment:
```suggestion
    await ensure_cancelled_many(update_tasks)
```
Use graceful cancel from golem-core.